src/main/java/baritone/pathing/movement/movements/MovementParkourAdv.java

```diff
@@ -0,0 +1,1054 @@
+                    lowestCost = getMoveResult(context, destX, srcY, destZ, extraAscend, posbJump, simpleDirection, type, placeCost, lowestCost, res);
+                    continue;
+                }
+            }
+        }
+        res = lowestCost;
+        if (res.cost < 1000) {
+            Vec3i jumpVec = new Vec3i(res.x - srcX, res.y - srcY, res.z - srcZ);
+            //System.out.println("Dir = " + simpleDirection + ", Cost: " + res.cost + ", Distance: " + getDistance(jumpVec, simpleDirection) + ", MoveDis: " + calcMoveDist(context, srcX, srcY, srcZ, extraAscend, jumpVec, simpleDirection));
+        }
+    }
+
+    private static MutableMoveResult getMoveResult(CalculationContext context, int x, int y, int z, double extraAscend, Vec3i jump, EnumFacing jumpDirection, JumpType type, double costModifiers, MutableMoveResult curLowestCost, MutableMoveResult res) {
+        res.x = x;
+        res.y = y;
+        res.z = z;
+        res.cost = costFromJump(context, x - jump.getX(), y - jump.getY(), z - jump.getZ(), extraAscend, jump, jumpDirection, type) + costModifiers;
+        if(res.cost < curLowestCost.cost) {
+            return res;
+        }
+        return curLowestCost;
+    }
+
+    private static double calcMoveDist(CalculationContext context, int srcX, int srcY, int srcZ, double extraAscend, Vec3i jump, EnumFacing jumpDirection) {
+        double ascendAmount = jump.getY() + extraAscend;
+
+        //Accounting for slab height
+        IBlockState destBlock = context.get(jump.getX() + srcX, jump.getY() + srcY - 1, + jump.getZ() + srcZ);
+        if(MovementHelper.isBottomSlab(destBlock)) {
+            ascendAmount -= 0.5;
+        }
+
+        jump = new Vec3i(jump.getX(), 0, jump.getZ());
+        double distance = getDistance(jump, jumpDirection);
+
+        //Modifying distance so that ascends have larger distances while descends have smaller
+        if(ascendAmount > 0) {
+            distance += ascendAmount * ASCEND_COST_PER_BLOCK;
+        } else {
+            distance += ascendAmount * -DESCEND_COST_PER_BLOCK; // ascendAmount is negative
+        }
+
+        //Calculating angle between vectors
+        Vec3d jumpVec = new Vec3d(jump.getX() - jumpDirection.getXOffset(), 0, jump.getZ() - jumpDirection.getZOffset()).normalize();
+        double angle = Math.acos(jumpVec.dotProduct(new Vec3d(jumpDirection.getDirectionVec()))); //in radians
+
+        //This distance is unitless as it contains: modifiers on ascends/descends, and the slowdowns in changing directions midair (angle)
+        return distance + TURN_COST_PER_RADIAN * angle;
+    }
+
+    @Override
+    public boolean safeToCancel(MovementState state) {
+        // since we don't really know anything about momentum, it can only be canceled during prep phase (i.e. before the jump)
+        // e.g. can't cancel while cancelling momentum or we may fall off the block.
+        return state.getStatus() != MovementStatus.RUNNING;
```

Comment:
why is space vs no space after `//` so inconsistent?

Reply:
I was used to having no spaces but I've seen other files in the project that have spaces so I was trying to add spaces but forgot sometimes. (sorry I'm new to programming)